### PR TITLE
Add minimum and maximum operations

### PR DESCRIPTION
**Pull request by stepanets**
add Lot interface
add MinOf, MaxOf and related
up interfaces dependency

### CODE REVIEW

File: src/Operations/MaxOf.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    public function __construct(Lot $lot, int $scale)
+    {
+        $this->lot = new EqualCurrenciesLot($lot);
+        $this->currency = new LotFirstCurrency($lot);
+        $this->scale = $scale;
+    }
+
+    public function amount(): string
+    {
+        $max = '';
```
Comment (tigokr) [inline src/Operations/MaxOf.php]: btw u can use first of a lot value

File: src/Operations/MaxOf.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        foreach ($this->lot->asIterator() as $money) {
+            if ('' === $max) {
+                $max = $money->amount();
+            } else {
+                $max = bccomp($max, $current = $money->amount(), $this->scale) < 0 ? $current : $max;
+            }
+        }
+
+        if ('' === $max) {
```
Comment (anikeevda) [inline src/Operations/MaxOf.php]: Maybe you should check size of lot first?

File: src/Operations/MinOf.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        $min = '';
+        foreach ($this->lot->asIterator() as $money) {
+            if ('' === $min) {
+                $min = $money->amount();
+            } else {
+                $min = bccomp($min, $current = $money->amount(), $this->scale) > 0 ? $current : $min;
+            }
+        }
+
+        if ('' === $min) {
```
Comment (anikeevda) [inline src/Operations/MinOf.php]: same here

File: src/JustMoney.php [hunk 4]
```diff
@@ -43,7 +44,7 @@ public function __construct(string $amount, Currency $currency, int $scale)
      */
     public function amount(): string
     {
-        return $this->amount;
+        return bcadd($this->amount, '0', $this->scale);
```
Comment (tigokr) [inline src/JustMoney.php:47]: why?

**Review by anikeevda — CHANGES_REQUESTED** [review]

File: src/Operations/MaxOf.php (hunk 2 above)
Comment (stepanets) [reply src/Operations/MaxOf.php]: This call evaluation of `asIterator()`

File: src/Operations/MinOf.php (hunk 3 above)
Comment (stepanets) [reply src/Operations/MinOf.php]: same answer

File: src/Operations/MaxOf.php (hunk 2 above)
Comment (stepanets) [reply src/Operations/MaxOf.php]: hmm, but I can call `valid()`

**Review by anikeevda — APPROVED** [review]

**Comment by tigokr** [comment]
whatever

File: src/JustMoney.php (hunk 4 above)
Comment (stepanets) [reply src/JustMoney.php:47]: without it returns the `$amount` as it was passed in the constructor, but I want to get `5.00` instead just '5' or '5.0000`, some sort of output unification

File: src/Operations/MaxOf.php (hunk 1 above)
Comment (stepanets) [reply src/Operations/MaxOf.php]: refactored

File: src/JustMoney.php (hunk 4 above)
Comment (tigokr) [reply src/JustMoney.php:47]: idk maybe another way existing?

Comment (stepanets) [reply src/JustMoney.php:47]: maybe, but that is the easiest

**Review by tigokr — APPROVED** [review]

**Pull request merged**